### DISCUSSION TITLE
Batch fix incorrect duration or CLI tool?

### DISCUSSION THREAD

**Discussion by Infinitay** [mifi/lossless-cut#1777] [Q&A]
Is it possible to do a batch fix incorrect durations or to call Lossless Cut via CLI in order to do so?

**Comment by mifi** [comment]
not currently possible but you  could run something like this:
```bash
for file in *.mp4; do
  ffmpeg -i "${file}" -map_metadata 0 -map 0 -ignore_unknown -c copy "${file}"-fixed.mp4
done
```

**Comment by mcp73** [comment]
> ```shell
> for file in *.mp4; do
>   ffmpeg -i "${file}" -map_metadata 0 -map 0 -ignore_unknown -c copy "${file}"-fixed.mp4
> done
> ```

Thank you for this and all you do! I can now run this as a script on all files before editing to save the step of fixing files individually.

One remaining issue... it copies both files with correct duration and incorrect duration. Would anyone be able to help with a conditional statement that first checks if the file needs to be fixed so that it only runs on incorrect files?

**Reply by Infinitay** [reply]
I'm not experienced with ffmpeg or ffprobe commands, but I'm not sure if there is a way to check to see if the file has an incorrect duration that's streamlined. I did a quick search and it seems like the current standard to do so on Windows is to output the errors to a file to see if there is any errors

**Reply by mcp73** [reply]
The same here. I did a quick search and looked through the source code, but didn't see any obvious solution. 

Maybe @mifi has a suggestion given that lossless-cut immediately detects incorrect duration upon loading the file. Hoping for a one-liner to put in an 'if' statement.

**Reply by mifi** [reply]
You can run ffprobe and check if the duration field is set to a valid value (on the format container afaik). I dont remember the exact command but it should be googleable
